src/threaddeadlock.h

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) 2019 Greg Griffith
+// Copyright (c) 2019 The Bitcoin Unlimited developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_THREAD_DEADLOCK_H
+#define BITCOIN_THREAD_DEADLOCK_H
+
+#include <boost/lexical_cast.hpp>
+#include <boost/thread.hpp>
+#include <inttypes.h>
+#include <map>
+#include <memory>
+#include <mutex>
+#include <set>
+#include <string>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "utilstrencodings.h"
+
+enum LockType
```

Comment:
Please add doxygen comments to all the external APIs in this whole file, so when we enable doxygen it will be part of it